.github/workflows/s3-rollout.yml

```
@@ -0,0 +1,90 @@
+---
+name: Rollout S3 buckets
+on:  # yamllint disable-line rule:truthy
+  workflow_call:
+    inputs:
+      # Scaleway information
+      s3-region:
+        description: Region to deploy to
+        required: true
+        type: string
+      s3-provider:
+        description: Provider to deploy to
+        required: true
+        type: string
+    secrets:
+      S3_ACCESS_KEY:
+        description: API key ID
+        required: true
+      S3_SECRET_KEY:
+        description: API key secret
+        required: true
+env:
+  # Default lifecycle.xml config
+  default-lifecycle: |
+    <LifecycleConfiguration>
+      <Rule>
+        <ID>Remove uncompleted uploads</ID>
+        <Filter>
+          <Prefix>/</Prefix>
+        </Filter>
+        <Status>Enabled</Status>
+        <AbortIncompleteMultipartUpload>
+          <DaysAfterInitiation>1</DaysAfterInitiation>
+        </AbortIncompleteMultipartUpload>
+      </Rule>
+    </LifecycleConfiguration>
+
+jobs:
+  generate-matrix:
+    name: Generate job matrices
+    runs-on: ubuntu-latest
+    outputs:
+      buckets: ${{ steps.filtered.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+            .github
+      - name: Generate Matrix
+        id: filtered
+        uses: dorny/paths-filter@v3
+        with:
+          filters: .github/s3-rollout-filters.yaml
+
+  rollout-buckets:
+    runs-on: ubuntu-latest
+    needs: [generate-matrix]
+    name: S3CMD
+    strategy:
+      matrix:
+        deployment: ${{ fromJson(needs.generate-matrix.outputs.buckets) }}
+    if: ${{ needs.generate-matrix.outputs.buckets != '[]' }}]
+    environment: ${{ matrix.deployment }}
+    steps:
+      - name: Checkout s3 folder
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+      - name: Set up S3cmd cli tool for environment
+        uses: s3-actions/s3cmd@main
+        with:
+          provider: ${{ inputs.s3-provider }}
+          region: ${{ inputs.s3-region }}
+          access_key: ${{ secrets.S3_ACCESS_KEY}}
+          secret_key: ${{ secrets.S3_SECRET_KEY }}
+      - name: Check existing/new buckets and apply lifecycle and cors
+        run: |
+          for e in s3/${{ matrix.deployment }}/*;
+          do
+            s3cmd mb s3://s3-$(basename -- $e)-${{ matrix.deployment }} || true
```

You may want to "calculate" the bucket name and store it in a variable, you're using it multiple times. So
```suggestion
            BUCKET_NAME="s3-$(basename -- $e)-${{ matrix.deployment }}"
            s3cmd mb s3://$BUCKET_NAME || true
```